File: 프로그래머스/level1_문자열 내림차순으로 배치하기.py
def solution(s):
    answer = ''
    s_list = list(s)
    s_list.sort(reverse=True)
    for value in s_list:
        answer += value
    #
    # for i in range(len(s_list)):
    #     if 65 <= ord(s_list[i]) <= 90:
    #         s_list.append(s_list[i])
    #     else:
    #         for j in range(i, len(s_list)):
    #             answer += s_list[j]
    #         break

    return answer

print(solution("Zbcdefg"))

File: 프로그래머스 복습(java)/src/level1/level1_문자열_내림차순으로_배치하기.java
package level1;

import java.util.*;

public class level1_문자열_내림차순으로_배치하기 {
    static class Solution {
        public String solution(String s) {
//            HashMap<Integer, Character> hashMap = new HashMap<>();
//            List<Integer> list = new ArrayList<>();
//            for (int i = 0; i < s.length(); i++) {
//                hashMap.put(s.charAt(i) - '0', s.charAt(i));
//                list.add(s.charAt(i) - '0');
//            }
//            list.sort(Comparator.reverseOrder());
//            StringBuilder answer = new StringBuilder();
//            for (int i = 0; i < s.length(); i++) {
//                answer.append(hashMap.get(list.get(i)));
//            }
//            return answer.toString();

            // More Simple and Better Code
            char[] sol = s.toCharArray();
            Arrays.sort(sol);
            return new StringBuilder(new String(sol)).reverse().toString();
        }
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        System.out.println(solution.solution("Zbcdefg"));
    }
}
